{{ cookiecutter.project_slug }}/.github/workflows/docs.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: pages
+  cancel-in-progress: false
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ "{{" }} steps.deployment.outputs.page_url {{ "}}" }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+{%- if cookiecutter.packaging == 'poetry' %}
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v3.0.0
+        with:
+          poetry-version: 1.8.3
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: poetry
+      - name: Install dependencies
+        run: poetry install
+      - name: Build docs with MkDocs
+        run: poetry run mkdocs build --strict
+{%- elif cookiecutter.packaging == 'pip-tools' %}
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: pip
+      - name: Install dependencies
+        run: pip install .[doc]
```

Comment:
This should use the `doc-requirements.txt`

Reply:
You mean we should do `pip install -r doc-requirements.txt` instead? What's the difference? Is it that it doesn't also install the contents of `requirements.txt`?

Reply:
I believe `pip install .[doc]` will not use the versions specified in `doc-requirements.txt`. So it will get latest versions of everything. Which isn't what is being tested for in the other ci file.